Create Terraform Script

HarithJ opened the pull request:
#### What does this PR do?
- Creates terraform script to spawn infrastructure on GKE
- Creates shell script that would create nginx ingress controller
- Gitignores .terraform folder and yml files inside ingress folder
- Adds documentation on how to spin up infrastructure using terraform

#### Description of Task to be completed?
https://www.pivotaltracker.com/story/show/163550391
https://www.pivotaltracker.com/story/show/163620828

#### How should this be manually tested?
1. Create a bucket on `andela-learning`
2. Open up `terraform/main.tf` file and substitute the bucket name with the bucket you just created.
3. Export GOOGLE_APPLICATION_CREDENTIALS environment variable with the value of the path to the GCP credentials.
4. Navigate to the `terraform` folder via terminal and run the following commands:
    - terraform init
    - terraform apply (terraform would ask you for the GCP project variable)
    This is it, the infrastructure is up and running.
5. Navigate to `ingress` folder via terminal
6. Connect to the cluster that terraform just created.
7. Reserve IP address on GCP
8. Export IP env var with the value of the IP address you just reserved.
9. Run ingress.sh script

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/163620828
https://www.pivotaltracker.com/story/show/163620828

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

HarithJ committed:
feat(terraform): create terraform script

- create terraform script to spawn infrastructure on GKE
- create shell script that would create nginx ingress controller
- gitignore .terraform folder and yml files inside ingress folder

[Delivers: #163550391]

akinsbo reviewed (changes requested):
Specify the provider version and create an Update.md file in a doc directory, to guide anyone planning to update the code on what exactly to update. It should state the current version of the provider.
Here is an example https://github.com/huginn/huginn/blob/master/doc/manual/update.md

HarithJ commented:
> update the code

I would like to suggest a better alternative to this part of your comment:
```create an Update.md file in a doc directory```

I think the best place to document `how to setup infrastructure using terraform` would be in Readme.md, because anytime I would like to “update the code” I don’t need to use terraform.

Ow, by the way, maybe if you are talking of updating the infrastructure, like for example, from `2 nodes` to `3 nodes`,  then maybe, but still I don’t think this would be necessary.

akinsbo commented:
All the information for updating `versions` of tools used in your infrastructure could be placed in one section of your readme.md or in a file `update.md`. So periodically, the entire codebase could be updated using that guide, then tested before being deployed.

If in future, your readme takes over three seconds to scroll down, consider breaking the update section into an `update.md` file

HarithJ committed:
chore(terraform): document how to use terraform scripts

- document how to use terraform scripts to spin up infrastructure on GKE
- document on updating google provider's version

[Delivers: #163550391]

mirr254 commented on terraform/gke/variables.tf:
```diff
@@ -0,0 +1,4 @@
+variable "project" {}
```
Since these variables don't have default values, have you included how we can use `.tfvars` file for such values? I can't seem to find it anywhere on your documentation.

HarithJ replied:
I have documented one way of supplying this variable, please have a look.

mirr254 replied:
could you please show me where.

mirr254 commented on terraform/main.tf:
```diff
@@ -0,0 +1,21 @@
+variable "project" {}
```
Same as mentioned above

mirr254 reviewed (approved):
LGTM
